routing/result_interpretation.go

```diff
@@ -517,10 +517,24 @@ func (i *interpretedResult) processPaymentOutcomeIntermediate(route *mcRoute,
 		if introIdx == len(route.hops.Val)-1 {
 			i.finalFailureReason = &reasonError
 		} else {
-			// If there are other hops between the recipient and
-			// introduction node, then we just penalize the last
-			// hop in the blinded route to minimize the storage of
-			// results for ephemeral keys.
+			// We penalize the final hop of the blinded route which
+			// is sufficient to not reuse this route again and is
+			// also more memory efficient because the other hops
+			// of the blinded path are ephemeral and will only be
+			// used in conjunction with the final hop. Moreover we
+			// don't want to punish the introduction node because
+			// the blinded failure does not necessarily mean that
+			// the introduction node was at fault. Penalizing the
+			// final hop does work here because we add a dummy hop
+			// when sending to a blinded route to make the path
+			// finding MPP compatible.
```

Comment:
nit: that part from `Penalizing` on looks contradictory to the above statement about the last hop before. I think it isn't necessary, because it's a detail in the pathfinder?

Reply:
removed